### ISSUE TITLE
MyPy & Pandas Issue with dtype in `read_sql_query`

### ISSUE THREAD

**Issue by bh2smith** [cowprotocol/dune-sync#24]
# MyPy "Bug"

I believe there is a bug in mypy on the dtype field. 
They expect types that I believe match what we pass, 
but also don't provide a public interface to import their types (also np.int64 doesn't work either).

```
src/fetch/orderbook.py:51: error: Argument "dtype" to "read_sql_query" has incompatible type "Dict[str, str]"; expected "Optional[Union[Union[ExtensionDtype, Union[str, dtype[generic], Type[str], Type[complex], Type[bool], Type[object]]], Dict[Any, Union[ExtensionDtype, Union[str, dtype[generic], Type[str], Type[complex], Type[bool], Type[object]]]]]]"  [arg-type]
```
We ignore types on this line only for now.

_Originally posted by @bh2smith in https://github.com/cowprotocol/dune-sync/pull/23#discussion_r1072095905_

### Additional context:

When attempting to import expected type declarations, like DType, we are faces with private interface pandas._typing

# Pandas "Bug"

Furthermore, the type issue arises from concatenating two similar data frames (one which is empty). 

1. If the dataframe has two fields (one integer and one float)
2. One of the two data frames is empty (so that without specifying types) the emptiness defaults to "object" type
3. After concatenation of the data frames, the float field remains float, but the integer field becomes "object"

**Comment by bh2smith** [comment]
For the pandas issue: Unfortunately I have not been able to reproduce naively with

```
    def test_df_concat_bug(self):
        df_1 = pd.DataFrame(
            {
                "integer_field": [16000018, 16000050],
                "float_field": [40.70410, 39.00522],
            }
        )
        df_2 = pd.DataFrame({})
        combined_12 = pd.concat([df_1, df_2])
        combined_21 = pd.concat([df_2, df_1])
        print(combined_12.dtypes)
        print(combined_21.dtypes)
```

**Comment by gentrexha** [comment]
I believe the issue is that when creating an empty dataframe the columns are float64 dtype:

```python
>>> df_2 = pd.DataFrame(
...         {
...             "integer_field": [],
...             "float_field": [],
...         }
...     )
>>> df_2.dtypes
integer_field    float64
float_field      float64
dtype: object
```

**Comment by bh2smith** [comment]
This was fixed in #46.
